exporter/collector/googlemanagedprometheus/naming.go

```diff
@@ -60,35 +61,24 @@ func (c Config) GetMetricName(baseName string, metric pmetric.Metric) (string, e
 // "/unknown" (eg, for Gauge) or "/unknown:{secondarySuffix}" (eg, "/unknown:counter" for Sum).
 // It also removes the "_total" suffix on an unknown counter, if this suffix was not present in
 // the original metric name before calling prometheus.BuildCompliantName(), which is hacky.
-// It is based on the untyped_prometheus_metric data point attribute, and behind a feature gate.
-func getUnknownMetricName(points pmetric.NumberDataPointSlice, suffix, secondarySuffix, originalName, compliantName string) string {
-	if !untypedDoubleExportFeatureGate.IsEnabled() {
-		return compliantName + suffix
-	}
-
+//
+//nolint:revive
+func getUnknownMetricName(points pmetric.NumberDataPointSlice, unknown bool, suffix, secondarySuffix, originalName, compliantName string) string {
```

Comment:
it would be nice if there was a clean way to just pass `metric` as a param (either instead of or along the `NumberDataPointSlice`), so then you could just call `isUnknown` inside the function. I'm assuming the bool param is what failed revive (the check is called something like conditional control..?)

Reply:
passed in metric directly